tests/pytests/test_cursors.py

```diff
@@ -314,9 +314,9 @@ def add_results(res):
         _, cursor = env.cmd('FT.AGGREGATE', 'idx', '*', 'LOAD', '*', 'WITHCURSOR', 'COUNT', count)
         env.cmd('FT.CURSOR', 'DEL', 'idx', cursor)
         # We expect that deleting the cursor will trigger the shards to delete their cursors as well.
-        # Since none of the cursors is expected to be expired, we don't expect `FT.CURSOR GC` to return a positive number.
-        # `FT.CURSOR GC` will return -1 if there are no cursors to delete, and 0 if the cursor list was empty.
-        env.expect('FT.CURSOR', 'GC', '42', '42').equal(0)
+        with TimeLimit(5, "shard cursors were not deleted"):
```

Comment:
Is this test reproducing the deadlock?

Reply:
Nope. Any test with cursors and FLUSHALL reproduces it (and we have many of them, even in this test).
This change was required because the `FT.CURSOR GC` call may be called before or after the `_FT.CURSOR DEL` command arrives, as it is async now